android-core/src/androidTest/java/com/mparticle/PushRegistrationTest.java

```diff
@@ -163,9 +163,17 @@ public void setPushRegistration(PushRegistrationHelper.PushRegistration pushRegi
                     //For enablePushNotifications() to set the push registration, we need to mimic
                     //the Firebase dependency, and clear the push-fetched flags
                     TestingUtils.setFirebasePresent(true, pushRegistration.instanceId);
-                    ConfigManager.getInstance(mContext).clearPushRegistration();
-
                     MParticle.getInstance().Messaging().enablePushNotifications(pushRegistration.senderId);
+                    //this method setting push is async, so wait for confirmation before continuing
+                    ConfigManager configManager = ConfigManager.getInstance(mContext);
+                    while (!configManager.isPushEnabled()) {
```

Comment:
wouldn't a listener as a callback be better than a while loop?  This is a test though so maybe it doesn't matter as much

Reply:
it would..but it's one of those things that isn't actually valuable to the client as far as I can tell... probably not worth going through the trouble of adding one right now.